ltr/client/opensearch_client.py

```diff
@@ -0,0 +1,263 @@
+import os
+import requests
+
+from .base_client import BaseClient
+from ltr.helpers.handle_resp import resp_msg
+
+import json
+from opensearchpy import OpenSearch
+from opensearchpy import helpers
+
+class ElasticResp():
+    def __init__(self, resp):
+        self.status_code = 400
+        if 'acknowledged' in resp and resp['acknowledged']:
+            self.status_code = 200
+        else:
+            self.status_code = resp['status']
+            self.text = json.dumps(resp, indent=2)
+
+class BulkResp():
+    def __init__(self, resp):
+        self.status_code = 400
+        if resp[0] > 0:
+            self.status_code = 201
+
+class SearchResp():
+    def __init__(self, resp):
+        self.status_code = 400
+        if 'hits' in resp:
+            self.status_code = 200
+        else:
+            self.status_code = resp['status']
+            self.text = json.dumps(resp, indent=2)
+
+
+class OpenSearchClient(BaseClient):
+    """ Note on the OpenSearch client,
+        OpenSearch LTR is not bound to an index like Solr LTR
+        so many calls take an index but do not use it
+
+        In the future, we may wish to isolate an Index's feature
+        store to a feature store of the same name of the index
+    """
+    def __init__(self, configs_dir='.'):
+        self.docker = os.environ.get('LTR_DOCKER') != None
+        self.configs_dir = configs_dir #location of elastic configs
+
+        if self.docker:
+            self.host = 'opensearch-node1'
```

Comment:
This would be `opensearch` when using the root docker-compose.yml?

Reply:
will update the root docker-compose.yml to be opensearch-node1. I've tested more with the notebooks/docker-compose.yml